src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -908,6 +917,10 @@ private void CheckModifiers(bool isExplicitInterfaceImplementation, Location loc
             {
                 diagnostics.Add(AccessCheck.GetProtectedMemberInSealedTypeError(ContainingType), location, this);
             }
+            else if (ContainingType is { IsExtension: true, ExtensionParameter: { Name: "" } } && !IsStatic)
```

Comment:
nit: this can be simplified
```suggestion
            else if (ContainingType is { IsExtension: true, ExtensionParameter.Name: "" } && !IsStatic)
```

Reply:
`ExtensionParameter` can be `null` even when `IsExtension` is `true`. That happens when the extension parameter is `__arglist`.

Reply:
Yes, but the pattern would handle `null`s right?

Reply:
Ah, you're right. I'll update